echo/index.js

```diff
@@ -181,15 +183,29 @@ function Client() {
     debug('request options', requestOptions);
 
     request.post(requestOptions, function onResp(err, response, body) {
-      var statusCode = response && response.statusCode
-        ? response.statusCode
-        : 0;
-
+      var statusCode = response && response.statusCode ? response.statusCode : 0;
       if (err || parseInt(statusCode / 100) !== 2) {
-        error('[echoClient] addEvents error', { err: err, body: body, statusCode: statusCode });
-        callback(err || 'error response status code: ' + statusCode);
+        var errorCode;
+
+        if (err) {
+          errorCode = errorCodesAndLabels.REQUEST_ERROR;
+        } else {
+          errorCode = httpStatusToErrorCode[statusCode] === undefined
+            ? errorCodesAndLabels.UNKNOWN_ERROR
+            : httpStatusToErrorCode[statusCode];
+        }
```

Comment:
I think this would read better as:

```js
if (err) {
    errorCode = errorCodesAndLabels.REQUEST_ERROR;
} else if (httpStatusToErrorCode[statusCode]) {
    errorCode = httpStatusToErrorCode[statusCode];
} else {
    errorCode = errorCodesAndLabels.UNKNOWN_ERROR;
}